site/source/docs/api_reference/val.h.rst

```diff
@@ -96,6 +96,43 @@ Guide material for this class can be found in :ref:`embind-val-guide`.
 
     Creates a ``val`` that represents ``null``.
 
+  .. _val_as_handle:
+  .. cpp:function:: EM_VAL as_handle() const
+    
+    Returns a raw handle representing this ``val``. This can be used for
+    passing raw value handles to JavaScript and retrieving the values on the
+    other side via ``emvalFromHandle`` function. Example:
+    
+    .. code:: cpp
+
+      EM_JS(void, log_value, (EM_VAL val_handle), {
+        var value = emvalFromHandle(val_handle);
+        console.log(value); // 42
+      });
+
+      val foo(42);
+      log_value(foo.as_handle());
+
+
+  .. cpp:function:: static val take_ownership(EM_VAL e)
+
+    Creates a ``val`` from a raw handle. This can be used for retrieving values
+    from JavaScript, where the JavaScript side should wrap a value with
+    ``emvalToHandle``, pass it to C++, and then C++ can use ``take_ownership``
+    to convert it to a ``val`` instance. Example:
+    
+    .. code:: cpp
+    
+      EM_ASYNC_JS(EM_VAL, fetch_json_from_url, (const char *url_ptr), {
+        var url = UTF8ToString(url);
+        var response = await fetch(url);
+        var json = await response.json();
+        return emvalToHandle(json);
```

Comment:
The name here surprised me, although I think I understand how it makes sense: The previous name `emvalFromHandle` can be read as "get an emval from a handle", and also as "emval namespace, method: from-handle". But this method name only makes sense in the second way, since it is not valid to read `emvalToHandle` as "turn an emval into a handle".

I think it's the word "to" that is ambiguous here. How about `emvalMakeHandle` or something like that?

Reply:
It really is hard...

I think you were right earlier though, we have very few namespaces, so that is actually a little odd to add.

Another idea: `jsToEmHandle, `emHandleToJS`?

Reply:
> we have very few namespaces

I mean, we do have `Asyncify` namespace... but what I meant is that we don't really mention `em` anywhere else despite all of those functions being Emscripten APIs.

> Another idea: jsToEmHandle, emHandleToJS?

Works for me! The only concern (aside from `em` mentioned above) I can think of is `js` being ambiguous on its own, but in context of conversions it shouldn't be.

Reply:
A downside to the namespace is that it is less DCE-friendly. That's the main reason we've been moving away from them, I believe. That is, it's not clear that closure always manages to remove unneeded parts of such "namespaces", although maybe we could check if that's the case here.

I'd still vote for `jsToEmVal / jsToEmHandle` myself, but I don't feel strongly.